packages/asset-swapper/src/quote_consumers/forwarder_swap_quote_consumer.ts

```diff
@@ -182,59 +182,50 @@ export class ForwarderSwapQuoteConsumer implements SwapQuoteConsumerBase<Forward
         const formattedFeePercentage = utils.numberPercentageToEtherTokenAmountPercentage(feePercentage);
-        try {
-            let txHash: string;
-            if (quoteWithAffiliateFee.type === MarketOperation.Buy) {
-                const { makerAssetFillAmount } = quoteWithAffiliateFee;
-                txHash = await this._contractWrappers.forwarder
-                    .marketBuyOrdersWithEth(
-                        orders,
-                        makerAssetFillAmount,
-                        orders.map(o => o.signature),
-                        formattedFeePercentage,
-                        feeRecipient,
-                    )
-                    .sendTransactionAsync({
-                        value,
-                        from: finalTakerAddress.toLowerCase(),
-                        gas: gasLimit,
-                        gasPrice,
-                    });
-            } else {
-                txHash = await this._contractWrappers.forwarder
-                    .marketSellOrdersWithEth(orders, orders.map(o => o.signature), formattedFeePercentage, feeRecipient)
-                    .sendTransactionAsync({
-                        value,
-                        from: finalTakerAddress.toLowerCase(),
-                        gas: gasLimit,
-                        gasPrice,
-                    });
-            }
-            return txHash;
-        } catch (err) {
-            if (_.includes(err.message, ContractError.SignatureRequestDenied)) {
-                throw new Error(SwapQuoteConsumerError.SignatureRequestDenied);
-            } else if (_.includes(err.message, ForwarderError.CompleteFillFailed)) {
-                throw new Error(SwapQuoteConsumerError.TransactionValueTooLow);
-            } else {
-                throw err;
-            }
+        let txHash: string;
+        if (quote.type === MarketOperation.Buy) {
+            const { makerAssetFillAmount } = quote;
+            txHash = await this._forwarder
+                .marketBuyOrdersWithEth(
+                    orders,
+                    makerAssetFillAmount,
+                    orders.map(o => o.signature),
+                    formattedFeePercentage,
+                    feeRecipient,
+                )
+                .sendTransactionAsync({
+                    from: finalTakerAddress,
+                    gas: gasLimit,
+                    gasPrice,
+                    value: providedEthAmount || ethAmountWithFees,
+                });
+        } else {
+            txHash = await this._forwarder
+                .marketSellOrdersWithEth(orders, orders.map(o => o.signature), formattedFeePercentage, feeRecipient)
+                .sendTransactionAsync({
+                    from: finalTakerAddress,
+                    gas: gasLimit,
+                    gasPrice,
+                    value: providedEthAmount || ethAmountWithFees,
+                });
         }
+        // TODO(dorothy-zbornak): Handle signature request denied
```

Comment:
I removed the `try/catch` that was here because it was doing nothing. `ContractError.SignatureRequestDenied` didn't actually get thrown by contract wrappers, and `sendTransactionAsync()` only mines the TX immediately on ganache, so you won't detect a failure on mainnet. 

`ContractError.SignatureRequestDenied` denied is easy enough to detect, but maybe the logic should live in another package. Detecting a `CompleteFillFailed` will have to be accomplished by doing a `callAsync()` at the same block the TX was mined. Let's add tasks for this.